src/gululog_idx.erl

```diff
@@ -264,15 +294,15 @@ truncate(Dir, #idx{tid = Tid, fd = Fd} = Idx, SegId, LogId, BackupDir) ->
   %% truncate idx file for = segid
   FileOpList2 = truncate_truncate_do(Dir, SegIdToTruncate, LogId, BackupDir),
   NewIdx =
-    %% verify the given logid is the first one
+    %% check if the given logid is the first one
     case LogId =:= ets:first(Tid) of
       true ->
         [] = wildcard_reversed(Dir), %% assert
         ok = close_cache(Tid),
         init(Dir);
       false ->
-        NewTid = truncate_cache(Tid, LogId),
-        {NewSegId, _} = locate_in_cache(Tid, LogId - 1),
+        NewSegId = get_segid(Tid, prev_logid(Tid, LogId)),
+        NewTid   = truncate_cache(Tid, mk_name(Dir, NewSegId), LogId),
```

Comment:
L298-L304
===>
```
case prev_logid(Tid, LogId) of
  false ->
    [] = wildcard_reversed(Dir), %% assert
    ok = close_cache(Tid),
    init(Dir);
  PrevLogId ->
    NewSegId = get_segid(Tid, PrevLogId),
```
It will reduce an operation of `ets`.